pkg/forwarder/transaction/sort_by_created_time_and_priority.go

```diff
@@ -0,0 +1,34 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package transaction
+
+import "sort"
+
+// SortByCreatedTimeAndPriority sorts transactions by creation time and priority
+type SortByCreatedTimeAndPriority struct {
+	HighPriorityFirst bool
+}
+
+// Sort sorts transactions by creation time and priority
+func (s SortByCreatedTimeAndPriority) Sort(transactions []Transaction) {
+	sorter := byCreatedTimeAndPriority(transactions)
+	if s.HighPriorityFirst {
+		sort.Sort(sorter)
+	} else {
+		sort.Sort(sort.Reverse(sorter))
+	}
+}
+
+type byCreatedTimeAndPriority []Transaction
```

Comment:
Type def should be at the top of the file. Also, all of the `*CreatedTimeAndPriority` vars/types/filename seem clunky and overly specific. What about something like `SortedTransactions` for types and `Sorter` for the file name?

Reply:
> Also, all of the *CreatedTimeAndPriority vars/types/filename seem clunky and overly specific. What about something like SortedTransactions for types and Sorter for the file name?

Sort by created time and priority is not an obvious way to sort so I think keeping an explicit name is better here.